src/persist-client/src/internal/machine.rs

```diff
@@ -646,9 +640,12 @@ where
         (seqno, maintenance)
     }
 
-    pub async fn maybe_become_tombstone(&mut self) -> Option<RoutineMaintenance> {
+    pub async fn become_tombstone(&mut self) -> Result<RoutineMaintenance, InvalidUsage<T>> {
         if !self.applier.since_upper_both_empty() {
```

Comment:
nit: lets have `since_upper_both_empty` return the error instead and just pass it on here. we shouldn't have any races between the two applier calls, but it's easy to combine them all under one lock

Reply:
Sorry, I don't quite follow the suggestion: were you thinking we'd keep the separate `since_upper_both_empty` call here and have that return a `Result<(),InvalidUsage>` / similar, or that we eliminate the method/call and do the check when we're holding the lock for the apply-unbatched-command below?

Reply:
Was suggesting the former, basically as an alternative removing `cfg(test)` from `fn since`. I think the latter would be even better, but it's a bigger change and we should hold off on that until we figure out the "one big write" fix

Reply:
Ah, I see which two applier calls you mean. That's better, thanks!